packages/turbo/src/common.ts

```diff
@@ -18,16 +11,23 @@ export enum SessionEvents {
   MESSAGE = 'MESSAGE',
 }
 
-const eventEmitter = new NativeEventEmitter(
-  NativeModules.RNVisitableViewModule
-);
-
 export function getNativeComponent<T>(componentName: string) {
-  return UIManager.getViewManagerConfig(componentName) != null
-    ? requireNativeComponent<T>(componentName)
-    : () => {
-        throw new Error(LINKING_ERROR);
-      };
+  const viewConfig = UIManager.getViewManagerConfig(componentName);
+
+  if (!viewConfig) {
+    throw new Error(LINKING_ERROR);
+  }
+
+  return {
+    [componentName]: requireNativeComponent<T>(componentName),
+    dispatchCommand: (ref, command, ...args) => {
+      UIManager.dispatchViewManagerCommand(
+        findNodeHandle(ref.current),
+        viewConfig.Commands[command],
+        args
+      );
+    },
+  };
```

Comment:
New way of native -> js communication. We dispatch an event using UIManager and catch it in the native view manager.